xbmc/GUIPassword.cpp

```diff
@@ -87,6 +90,9 @@ bool CGUIPassword::IsItemUnlocked(T pItem,
         CMediaSourceSettings::GetInstance().UpdateSource(strType, strLabel, "badpwdcount",
                                                          std::to_string(pItem->m_iBadPwdCount));
         CMediaSourceSettings::GetInstance().Save();
+
+        // refresh favourites due to possible visibility changes
```

Comment:
The comment should contain why this needs to be called here. Not a general reason but the specific reason.